R/sidebar.R

```diff
@@ -130,7 +133,7 @@ sidebar <- function(
         title = "Toggle sidebar",
         "aria-expanded" = if (open %in% c("open", "desktop")) "true" else "false",
         "aria-controls" = id,
-        collapse_icon()
+        collapse_icon %||% sidebar_collapse_icon()
```

Comment:
We should add the `aria-label` attribute too 

```patch
        title = "Toggle sidebar",
+       "aria-label" = "Toggle sidebar",
        "aria-expanded" = if (open %in% c("open", "desktop")) "true" else "false",
        "aria-controls" = id,
        collapse_icon %||% sidebar_collapse_icon()
```

or if you add `collapse_label` you could reuse that for `title` and `aria-label`.

```patch
-       title = "Toggle sidebar",
+       title = collapse_label,
+       "aria-label" = collapse_label,
        "aria-expanded" = if (open %in% c("open", "desktop")) "true" else "false",
        "aria-controls" = id,
        collapse_icon %||% sidebar_collapse_icon()
```

Reply:
Given that `title`/`aria-label` has the same value here, it feels a bit strange to me that this would provide anything of real value. Can you elaborate on how the difference would be noticable (I tested VoiceOver with just the title and the announcement seemed sensible to me)

Reply:
ChatGPT gave me this explanation, which is good enough for me:

>  The title attribute is a standard HTML attribute used to provide additional information about an element when the user hovers their mouse pointer over it. It is primarily used for providing supplementary information to sighted users. It's not a recommended method for conveying information to screen readers, as it's not consistently supported and may not be announced by all screen readers.

Reply:
Yeah `title` isn't consistently well-handled by screen readers, and `aria-label` is required if there isn't any text content inside the button (see bottom of https://www.w3.org/WAI/ARIA/apg/patterns/button/).